docs/index.rst

```diff
@@ -73,6 +73,9 @@ or ``theme.conf`` for more details.
     mermaid_diagrams
     task_lists
     keys
+    format_signatures
+    cpp
+    external_cpp_references
```

Comment:
I think the domain customizations should be part of a separate `toctree` in the docs (may include the formatting extension doc also).